code/mission/missiongoals.cpp

```diff
@@ -978,10 +978,11 @@ void mission_process_event( int event )
 	}
 
 	// decrement the trigger count.  When at 0, set the repeat count to 0 so we don't eval this function anymore
-	if (result && (Mission_events[event].trigger_count > 0) && (Mission_events[event].flags & MEF_USING_TRIGGER_COUNT) ) {
-		Mission_events[event].trigger_count--;
+	if (result && (Mission_events[event].trigger_count != 0) && (Mission_events[event].flags & MEF_USING_TRIGGER_COUNT) ) {
+		if (Mission_events[event].trigger_count > 0)
```

Comment:
How is this change related to ship_info vectorization? There is another one below that also doesn't look like it's related.